File: admin/src/api/feedback.ts
import { client } from './client'
import type { ApiResponse, PagedResponse } from '../types/api'

export interface Feedback {
  id: number
  type: 'bug' | 'feature' | 'other'
  title: string
  content: string
  contact: string
  app_version: string
  os: string
  created_at: string
}

export interface FeedbackFilter {
  type?: string
  start_date?: string
  end_date?: string
}

export function listFeedbacks(page = 1, pageSize = 20, filter: FeedbackFilter = {}) {
  let url = `/feedbacks?page=${page}&page_size=${pageSize}`
  if (filter.type) {
    url += `&type=${encodeURIComponent(filter.type)}`
  }
  if (filter.start_date) {
    url += `&start_date=${encodeURIComponent(filter.start_date)}`
  }
  if (filter.end_date) {
    url += `&end_date=${encodeURIComponent(filter.end_date)}`
  }
  return client.get<ApiResponse<PagedResponse<Feedback>>, ApiResponse<PagedResponse<Feedback>>>(url)
}

export function deleteFeedback(id: number) {
  return client.delete<ApiResponse, ApiResponse>(`/feedbacks/${id}`)
}

File: server/internal/domain/feedback.go
package domain

import (
	"time"
)

type Feedback struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	Type       string    `gorm:"size:20;not null" json:"type"`
	Title      string    `gorm:"size:200;not null" json:"title"`
	Content    string    `gorm:"size:5000;not null" json:"content"`
	Contact    string    `gorm:"size:200" json:"contact"`
	AppVersion string    `gorm:"size:50" json:"app_version"`
	OS         string    `gorm:"size:50" json:"os"`
	IP         string    `gorm:"size:45" json:"-"`
	UserAgent  string    `gorm:"size:500" json:"-"`
	CreatedAt  time.Time `json:"created_at"`
}

func (Feedback) TableName() string {
	return "feedbacks"
}
